Automattic/wp-calypso
# Hosting configuration: poll Atomic transfer status when loading the page (v2)

Closes https://github.com/Automattic/wp-calypso/issues/72371
Related https://github.com/Automattic/wp-calypso/pull/77112 (I ported this original PR to the latest trunk and made some small improvements along the way)

## Proposed Changes

* Make sure the `/hosting-config` UI updates correctly when the Atomic transfer completes ( see https://github.com/Automattic/wp-calypso/issues/72371)
* When transferring, make the client-side status polling mechanism robust enough to survive a hard page refresh. 
* Now that the polling mechanism can pick up existing jobs, there does not seem to be a good reason to lock them into this page until the job is complete. I have removed the "click outside" logic in this PR. We still might want to include this. I would like @aidvu's input here, as they added the original restriction. 

## Testing Instructions



There's been quite a few changes to boolean logic so there are two main things to test:

1. The state of `/hosting-config` for various plan + site type combos
2. The robustness of the Atomic transfer mechanism. See https://github.com/Automattic/wp-calypso/issues/72371 for inspiration. In short, it should be fine to go to another page and come back or do a hard refresh. The process should always complete and not get stuck. 

First, try this feature on prod and see that the process is broken and gets stuck. In my tests, it completed occasionally but mostly did not until I refreshed the page.  

Next, test viewing `/hosting-config` when you have:
 
**A Free site**: 
       * should see the upsell
       * should not see the activation banner
       * features should be masked

![image](https://github.com/Automattic/wp-calypso/assets/6851384/636bc92d-7375-485a-a8a7-2e068253baaf)
    
**a Creator plan but haven't gone Atomic**
       * You should see an activation banner but no upsell
       * Feature cards should be masked and unusable

![image](https://github.com/Automattic/wp-calypso/assets/6851384/1a7104f2-f745-499b-9a83-34c600b29878)
 
**an Atomic site**:
        * upsell should be gone
        * The activation banner should be gone 
        * All features should be unmasked and usable
        
<img width="1145" alt="Screenshot 2567-01-05 at 13 56 12" src="https://github.com/Automattic/wp-calypso/assets/6851384/a9e60841-87c5-4230-90e7-fb4b289a81db">
      
* **Test upgrading a Free plan to Creator**
       * should initially be like **A Free site** above
       * Then, when the process completes, it should be like **an Atomic site** above

* General robustness. We are polling the server for updates and should handle a refresh. Do your best to try and break it. 

[screen-capture (9).webm](https://github.com/Automattic/wp-calypso/assets/6851384/3c4e1a89-680f-404d-8f09-bf18021ccab0)


## Pre-merge Checklist
